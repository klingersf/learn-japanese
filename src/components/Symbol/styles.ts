import styled from 'styled-components'

export const Button = styled.div`
  cursor: pointer;
  background: #f231a5;
  border: none;
  color: white;
  padding: 15px 30px;
  border-radius: 5px;
  box-shadow: 4px 4px 0px #be0374;
  font-size: 30px;
  transition: box-shadow 0.5s;

  &:hover {
    box-shadow: none;
  }
`
