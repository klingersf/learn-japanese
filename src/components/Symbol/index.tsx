import * as S from './styles'

export type SymbolProps = {
  japanese: string
  english: string
  speak: () => void
}

const Symbol = ({ japanese, english, speak }: SymbolProps) => (
  <S.Button onClick={speak}>
    <h1>{japanese}</h1>
    <p>{english}</p>
  </S.Button>
)

export default Symbol
