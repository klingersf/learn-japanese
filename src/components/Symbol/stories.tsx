import { Story, Meta } from '@storybook/react/types-6-0'
import Symbol, { SymbolProps } from '.'

export default {
  title: 'Symbol',
  component: Symbol,
  args: {
    japanese: '„ÅÇ',
    english: 'a'
  }
} as Meta

export const Default: Story<SymbolProps> = (args) => <Symbol {...args} />
