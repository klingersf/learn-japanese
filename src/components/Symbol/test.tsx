import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import Symbol from '.'

describe('<Symbol />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <Symbol japanese="あ" english="a" speak={jest.fn} />
    )

    expect(screen.getByRole('heading', { name: /あ/i })).toBeInTheDocument()
    expect(screen.getByText(/a/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should call speak function when click', () => {
    const speak = jest.fn()
    render(<Symbol japanese="あ" english="a" speak={speak} />)

    userEvent.click(screen.getByText(/あ/i))

    expect(speak).toHaveBeenCalled()
  })
})
