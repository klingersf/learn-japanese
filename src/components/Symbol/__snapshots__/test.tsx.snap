// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Symbol /> should render the heading 1`] = `
.c0 {
  cursor: pointer;
  background: #f231a5;
  border: none;
  color: white;
  padding: 15px 30px;
  border-radius: 5px;
  box-shadow: 4px 4px 0px #be0374;
  font-size: 30px;
  -webkit-transition: box-shadow 0.5s;
  transition: box-shadow 0.5s;
}

.c0:hover {
  box-shadow: none;
}

<div
  class="c0"
>
  <h1>
    „ÅÇ
  </h1>
  <p>
    a
  </p>
</div>
`;
